platform/src/main/java/org/stellar/anchor/platform/event/ClientStatusCallbackHandler.java

```diff
@@ -64,13 +70,14 @@ public ClientStatusCallbackHandler(
     this.clientConfig = clientConfig;
     this.assetService = assetService;
     this.sep6TransactionStore = sep6TransactionStore;
+    this.customerIntegration = customerIntegration;
     this.sep6MoreInfoUrlConstructor = sep6MoreInfoUrlConstructor;
     this.sep24MoreInfoUrlConstructor = sep24MoreInfoUrlConstructor;
   }
 
   @Override
   boolean handleEvent(AnchorEvent event) throws IOException {
-    if (event.getTransaction() != null) {
+    if (event.getTransaction() != null || event.getCustomer() != null) {
```

Comment:
Where is this event fired?

Reply:
https://github.com/stellar/java-stellar-anchor-sdk/blob/3e6dc69754556e40400c03eac3992e48022c5fb8/core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java#L112-L119

Reply:
From my understanding it's only called if SEP12 was called. I think it's no longer a correct place to fire an event, we should do it from the RPC call. 
Because otherwise events won't be fired when anchor updates from PROCESSING to other status

Reply:
> only expected to make notify_customer_info_updated RPC calls when the customer status updates asynchronously outside of a request/response cycle

No, I don't think that's the case. If you don't call the RPC, Platform doesn't know about status change. In the file Philip linked above we do the callback, but we only receive id in response. We simply fire an event and that's it. There's no check for kyc status, and there will be no update done to the transaction status.

Reply:
That makes sense. The wallet doesn't need to know about the updates it initiates; it only needs to know when the business server has updated its evaluation of the customer status, which is made known to the AP through this RPC.

Reply:
We need to keep firing the event here in case anyone depends on it. We should remove it in 3.0.

Reply:
I see. So if a business wants to provide SEP-12 callbacks, they need to call `notify_customer_info_updated` for every customer status change, even if that status change happened synchronously as a result of a wallet's `PUT /customer` request.

I feel like there is an opportunity to improve this.

If the anchor platform is going to call `GET /customer` as a result of receiving a `notify_customer_info_updated` request, should it also make a `GET /customer` as a result of a wallet's `PUT /customer` request as well? 

That way the business only uses `notify_customer_info_updated` when there is an update outside of a request/response cycle. The only time that happens is when a customer goes from `PROCESSING` to another status.

Or, we could update the SEP-12 protocol to return the full customer response body from `PUT /customer` requests, negating the need to send a callback.

Reply:
Business should just return full body customer response in `PUT /customer` IMO

Reply:
> If the anchor platform is going to call GET /customer as a result of receiving a notify_customer_info_updated request, should it also make a GET /customer as a result of a wallet's PUT /customer request as well?
>
> That way the business only uses notify_customer_info_updated when there is an update outside of a request/response cycle. The only time that happens is when a customer goes from PROCESSING to another status.

It's a bit awkward making a GET after the PUT but that's my plan.

Reply:
@philipliu do you mind creating a ticket to implement this functionality in 3.x?